File: solutions/c/darts/1/darts.c
#include "darts.h"
#include <math.h>

uint8_t score(coordinate_t landing_position) {
    // Calculate squared distance from center
    float squared_distance = landing_position.x * landing_position.x 
                           + landing_position.y * landing_position.y;
    
    // Compare squared distances to squared radii
    if (squared_distance <= 1.0f * 1.0f) {      // radius^2 = 1
        return 10;
    } else if (squared_distance <= 5.0f * 5.0f) {  // radius^2 = 25
        return 5;
    } else if (squared_distance <= 10.0f * 10.0f) { // radius^2 = 100
        return 1;
    } else {
        return 0;
    }
}

File: solutions/python/darts/1/darts.py
import math

def score(x, y):
    """
    Calculate the score of a dart landing at position (x, y).
    
    Scoring rules:
    - Outside target (radius > 10): 0 points
    - Outer circle (5 < radius <= 10): 1 point
    - Middle circle (1 < radius <= 5): 5 points
    - Inner circle (radius <= 1): 10 points
    """
    # Calculate distance from center (0, 0)
    distance = math.sqrt(x**2 + y**2)
    
    # Determine score based on distance
    if distance > 10:
        return 0
    elif distance > 5:
        return 1
    elif distance > 1:
        return 5
    else:
        return 10